R/epidist_db.R

```diff
@@ -0,0 +1,96 @@
+#' Create an `epidist` object directly from the epiparameter library (database)
+#'
+#' @description This function can extract an `epidist` object directly from the
+#' library of epidemiological parameters without having to read in an `epiparam`
+#' object and pull out an `epidist` object from one of the entries. If a
+#' distribution from a specific study is required, the `author` argument can
+#' be specified. For now this must match the author entry in the database
+#' exactly to be recognised.
+#'
+#' @details The current criteria for chosing a distribution when multiple
+#' entries exist in the database for a given disease and epidemiological
+#' distribution is to return the entry with the largest sample size.
+#'
+#' @param disease A character string specifying the disease
+#' @param epi_dist A character string specifying the epidemiological
+#' distribution
+#' @param author The author of the study reporting the distribution
+#'
+#' @return An `epidist` object
+#' @export
+#'
+#' @examples
+#' epidist_db(disease = "influenza", epi_dist = "serial_interval")
+#'
+#' # comparison between using `epidist_db()` and `epiparam()` with
+#' # `as_epidist()`
+#'
+#' # load influenza serial interval from database
+#' edist <- epidist_db(disease = "influenza", epi_dist = "serial_interval")
```

Comment:
Note that when printing a summary of `epi_dist_db` in R, `hospitalisation_to_death` is at the top of list as parameter, but not supported yet as `epidist` object I think?

Reply:
`<epidist>` should support `hospitalisation_to_death`. Here is an example where it seems to work fine:

``` r
library(epiparameter)
epi_dist_db <- epiparam()
epi_dist_db
#> Epiparam object
#> Number of distributions in library: 118
#> Number of diseases: 23
#> Number of delay distributions: 95
#> Number of offspring distributions: 10
#> Number of studies in library: 57
#> <Head of library>
#>       disease         epi_distribution prob_distribution
#> 1  Adenovirus        incubation_period             lnorm
#> 2 Chikungunya        incubation_period             lnorm
#> 3    COVID-19 hospitalisation_to_death           weibull
#> 4    COVID-19 hospitalisation_to_death             lnorm
#> 5    COVID-19        incubation_period              <NA>
#> 6    COVID-19        incubation_period           weibull
#> <112 more rows & 53 more cols not shown>
edist <- as_epidist(epi_dist_db[3,])
#> Using Linton et al. (2020) <10.3390/jcm9020538> PMID: 32079150. 
#> To retrieve the short citation use the 'get_citation' function
#> Numerical approximation used, results may be unreliable.
edist
#> Disease: COVID-19
#> Pathogen: SARS-CoV-2
#> Epi Distribution: hospitalisation to death
#> Study: Linton et al. (2020) <10.3390/jcm9020538> PMID: 32079150
#> Distribution: weibull
#> Parameters:
#>   shape: 1.599
#>   scale: 9.926
```

<sup>Created on 2023-05-30 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Please let me know if there is an issue somewhere else, and I'll make an issue and resolve.